src/components/tabset/tabset.component.spec.ts

```diff
@@ -116,12 +120,62 @@ describe('Tabset Component', () => {
         expect(component.onTabDeselect).toHaveBeenCalled();
     });
 
+    it('should change to new tab when tab is programmatically selected by tab index', () => {
+
+        const tab1 = getTabForClass(0, nativeElement);
+        const tab3 = getTabForClass(2, nativeElement);
+
+        // check first tab is active
+        expect(tab1.classList.contains('active')).toBeTruthy();
+
+        const tabButton: HTMLElement = nativeElement.querySelector('.button-primary');
+        expect(tabButton).toBeTruthy();
+
+        spyOn(component, 'onTabActivated');
+
+        // programatically change tab by index
+        tabButton.click();
```

Comment:
Since this is a unit test, you can simplify this by just calling `component.selectTab(...)`. We don't need any buttons.